src/Authentik/Provider.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+namespace SocialiteProviders\Authentik;
+
+use SocialiteProviders\Manager\OAuth2\AbstractProvider;
+use SocialiteProviders\Manager\OAuth2\User;
+
+class Provider extends AbstractProvider
+{
+    /**
+     * Unique Provider Identifier.
+     */
+    const IDENTIFIER = 'AUTHENTIK';
+
+    /**
+     * {@inheritdoc}
+     */
+    protected $scopes = ['openid goauthentik.io/api profile email'];
+
+    /**
+     * {@inheritdoc}
+     */
+    public static function additionalConfigKeys()
+    {
+        return ['base_url'];
+    }
+
+    protected function getBaseUrl()
+    {
+        return rtrim($this->getConfig('base_url'), '/');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getAuthUrl($state)
+    {
+        return $this->buildAuthUrlFromBase($this->getBaseUrl().'/application/o/authorize/', $state);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getTokenUrl()
+    {
+        return $this->getBaseUrl().'/application/o/token/';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getUserByToken($token)
+    {
+        $response = $this->getHttpClient()->get($this->getBaseUrl().'/application/o/userinfo/', [
+            'headers' => [
+                'Authorization' => 'Bearer '.$token,
+            ],
+        ]);
+
+        return json_decode($response->getBody(), true);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function mapUserToObject(array $user)
+    {
+        return (new User())->setRaw($user)->map([
+            'email'              => $user['email'] ?? '',
```

Comment:
See `\Laravel\Socialite\Contracts\User` for the required fields
You need at mminima an email, and id or a nickname to identify the user.

check for example https://github.com/laravel/socialite/blob/5.x/src/Two/GoogleProvider.php#L72-L78

Reply:
Am I missing something?
In the part of the code you are referring an email and a nickname are already set

Reply:
Both email and nickname can be empty string.
And no user id is provided.

If you don't provide a unique user id, you need to provide some non-empty unique strings somewhere, right ?

Reply:
I've adjusted the returned values to return the sub as an id field. That will never be empty (it's the subject of the authentication)